dotnet/src/Connectors/Connectors.Amazon/Bedrock/Models/BedrockUtilities.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Amazon.BedrockRuntime;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Connectors.Amazon.Models;
+
+/// <summary>
+/// Utilities class for functions all Bedrock models need to use.
+/// </summary>
+public class BedrockUtilities
+{
+    /// <summary>
+    /// Maps the AuthorRole to the corresponding ConversationRole because AuthorRole is static and { readonly get; }.
+    /// </summary>
+    /// <param name="role"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    public ConversationRole MapRole(AuthorRole role)
+    {
+        if (role == AuthorRole.User)
+        {
+            return ConversationRole.User;
+        }
+
+        if (role == AuthorRole.Assistant)
+        {
+            return ConversationRole.Assistant;
+        }
+        throw new ArgumentOutOfRangeException(nameof(role), $"Invalid role: {role}");
```

Comment:
I had the same thought. The issue is that the `ChatHistory` object that the Semantic Kernel has users create for chat completion needs to be converted to a `ConverseRequest` object (to run Bedrock's `ConverseAsync`). `ConverseRequest` takes `List<Message> Messages` which where each `Message` object has 
```
List<ContentBlock> _content
ConversationRole _role;
```
and `ConversationRole` only has user and assistant. Meanwhile `ChatHistory` has `List<ChatMessageContent> _messages` where each `ChatMessageContent` has `AuthorRole Role` and `AuthorRole` has system, assistant, user, and tool.

Reply:
Right, but for example `IChatCompletionRequest` has `public List<SystemContentBlock>? System { get; set; }` - which if I understand correctly is for keeping system prompt, so probably correct solution would be skip system messages during mapping to `Messages` and add them to `System`.

Reply:
Yes, that makes sense! I'm filtering out system messages (and tool messages for that matter) from the mapping instead of throwing an error and adding it directly to the `ConverseRequest` `List<SystemContentBlock>` parameter now.